packages/nimbus/src/components/number-input/number-input.recipe.tsx

```diff
@@ -0,0 +1,169 @@
+  slots: ["root", "input", "incrementButton", "decrementButton"],
+
+  // Base styles applied to all instances of the component
+  base: {
+    root: {
+      display: "inline-block",
+      position: "relative",
+    },
+    input: {
+      display: "block",
+      borderRadius: "200",
+      colorPalette: "neutral",
+      focusVisibleRing: "outside",
+      bg: "transparent",
+      outline: "none",
+      boxShadow: "inset 0 0 0 var(--border-width) var(--border-color)",
+      appearance: "textfield",
+      width: "full",
+      _placeholder: {
+        opacity: 0.5,
+        color: "currentColor",
+      },
+      _disabled: {
+        layerStyle: "disabled",
+      },
+      "&[data-invalid='true']": {
+        "--border-color": "colors.critical.7",
+        color: "critical.11",
+      },
+    },
+    incrementButton: {
+      h: "50%",
+      display: "flex",
+      alignItems: "center",
+      justifyContent: "center",
+      cursor: "button",
+      bg: "transparent",
+      _hover: {
+        bg: "primary.2",
+      },
+      _active: {
+        bg: "neutral.4",
+      },
+      _disabled: {
+        opacity: 0.5,
+        cursor: "not-allowed",
+      },
+      "& svg": {
+        fill: "primary.9",
+      },
+      borderTopRightRadius: "200",
+      "&[data-invalid='true']": {
+        "--border-color": "colors.critical.7",
+      },
+    },
+    decrementButton: {
+      h: "50%",
+      display: "flex",
+      alignItems: "center",
+      justifyContent: "center",
+      cursor: "button",
+      bg: "transparent",
+      _hover: {
+        bg: "primary.2",
+      },
+      _active: {
+        bg: "neutral.4",
+      },
+      _disabled: {
+        opacity: 0.5,
+        cursor: "not-allowed",
+      },
+      "& svg": {
+        fill: "primary.9",
+      },
+      borderBottomRightRadius: "200",
+      "&[data-invalid='true']": {
+        "--border-color": "colors.critical.7",
+      },
+    },
+  },
+
+  variants: {
+    size: {
+      sm: {
+        input: {
+          h: 800,
+          textStyle: "sm",
+          px: 300,
+          paddingRight: 800,
+        },
+        incrementButton: {
+          w: 600,
+        },
+        decrementButton: {
+          w: 600,
+        },
+      },
+      md: {
+        input: {
+          h: 1000,
+          textStyle: "md",
+          px: 400,
+          paddingRight: 1000,
+        },
+        incrementButton: {
+          w: 600,
+        },
+        decrementButton: {
+          w: 600,
+        },
+      },
+    },
+
+    variant: {
+      solid: {
+        input: {
+          "--border-width": "sizes.25",
+          "--border-color": "colors.neutral.7",
+          backgroundColor: "neutral.1",
+          _hover: {
+            backgroundColor: "primary.2",
+          },
+        },
+        incrementButton: {
+          "--border-color": "colors.neutral.7",
+          boxShadow:
+            "inset 1px 0 0 0 var(--border-color), inset 0 -0.5px 0 0 var(--border-color)",
```

Comment:
@misama-ct For the life of me, I could not get `var(--border-width)` to work here instead of the standard `1px`, even though it works in other files. Am I missing something basic?